core/src/main/java/discord4j/core/object/entity/Message.java

```diff
@@ -263,13 +263,29 @@ public boolean mentionsEveryone() {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Gets the users with partial members specifically mentioned in this message, without duplication and with the same order
+     * as in the message.
+     *
+     * @return The users with partial members specifically mentioned in this message, without duplication and with the same order
+     * as in the message.
+     */
+    public List<UserWithMember> getUserAndMemberMentions() {
+        return data.mentions().stream()
+            .map(data -> new UserWithMember(gateway, data,
+                getGuildId().map(Snowflake::asLong).orElse(null)))
+            .collect(Collectors.toList());
```

Comment:
We should first check if `data.guildId()` is absent. If it is, we can just immediately return `Collections.emptyList()`. Otherwise, we can assume that `data.member()` (inside the `map()`) is present. This is what will allow us to combine `MemberPart` and `UserWithMember`. The "is the member data present?" question is answered here, rather than being put off until `getMember()`.